test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -0,0 +1,92 @@
+defmodule Trento.DeregistrationProcessManagerTest do
+  use ExUnit.Case
+
+  alias Trento.Domain.Events.{
+    HostDeregistered,
+    HostDeregistrationRequested,
+    HostRegistered,
+    HostRolledUp
+  }
+
+  alias Trento.DeregistrationProcessManager
+  alias Trento.Domain.Commands.DeregisterHost
+
+  describe "events interested" do
+    test "should start the process manager when HostRegisteredEvent arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%HostRegistered{host_id: host_id})
+    end
+
+    test "should start the process manager when HostRolledUp arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%HostRolledUp{host_id: host_id})
+    end
+
+    test "should continue the process manager when HostDeregistrationRequested arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:continue, ^host_id} =
+               DeregistrationProcessManager.interested?(%HostDeregistrationRequested{
+                 host_id: host_id
+               })
+    end
+
+    test "should stop the process manager when HostDeregistered arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:stop, ^host_id} =
```

Comment:
I'm not sure about: is the Process Manager stopped because it got never started in the first place or because the `%HostDeregistered` event arrived?

I would expect to see something explicit that causes the registration in the first place, am I missing something?

Reply:
Yes, the process manager is stopped when HostDeregistered is emitted, and started when HostRegistered or HostRolledUp is emitted

Reply:
But shouldn't the test have something starting the process manager first? (before stopping it with the `%HostDeregistered` event)

Reply:
Yes, you can look at the different match of the `interested?` function, we have some cases when we return :start, the process manager itself is started by commanded, it's just another genserver under the hood

Reply:
@rtorrero those are unit tests, here the test is to show that certain functions return expected output given certain inputs. 
The lifecycle of the ProcessManager is managed by commanded, so we don't need to start a process "for real", we just need to test that the business logic is correct.